### ISSUE TITLE
SetPackagePath broken by HPC-GAP library merges

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#268]
Commit 687508458b643c4a08624c5bb9eed8bde0278c39 broke `SetPackagePath` for me, probably because it made something immutable that wasn't before:
```
gap> SetPackagePath("anupq", ".");
Error, Records Assignment: <rec> must be a mutable record in
  GAPInfo.PackagesInfo.(pkgname) := [ record ]; on line 1677 of file /Users/mhorn/Projekte/GAP/gap.github/lib/package.gi called from
<function "SetPackagePath">( <arguments> )
 called from read-eval loop at line 2 of *stdin*
you can 'return;' and ignore the assignment
brk>
```

**Comment by fingolfin** [comment]
Specifically, the line 303 in package.gi, which was changed to
```diff
-    GAPInfo.PackagesInfo:= record;
+    GAPInfo.PackagesInfo:= `record;
```
A quick fix would be to just revert that particular line. But then we get a diff with HPC-GAP again. Which is not terrible for now, I guess. A "proper" fix may require copying the whole `GAPInfo.PackagesInfo` record, modify that one entry, then make the result immutable again. Quite ugly, but then, this is a very rare operation, so it doesn't need to be super-efficient.

**Comment by ChrisJefferson** [comment]
Another option (which I'm trying now) is to make GAPInfo.PackagesInfo an atomic record. This seems to be working OK (it requires making each member of PackagesInfo immutable, but I think they probably should be).

**Comment by alex-konovalov** [comment]
FYI, I made it an atomic record in d7252b74da152cd1f106f2cbaeabbdee3867d40a in 2011, and then @ChrisJefferson made it immutable in f247ed683608fd54d9e5af11399b7744064438f5 in 2014. I haven't yet found any traces of email communications about that though.

**Comment by ChrisJefferson** [comment]
I made it immutable, because although it was atomic, the members where still thread-local, so couldn't be read off the main thread. Making the whole thing immutable was the quickest solution to this, I didn't notice it got changed (sorry).

So, we want it back to AtomicRecord, and then explictly make all the things in it immutable.

**Comment by alex-konovalov** [comment]
@ChrisJefferson I agree. If we make it an atomic record with components being immutable objects, then things like SetPackagePath also needs modification if they need to change these objects.

**Comment by ChrisJefferson** [comment]
Fixed in #271 (hopefully)

**Comment by alex-konovalov** [comment]
@fingolfin are you happy with the fix? If so, could you please close this issue.

**Comment by alex-konovalov** [comment]
Fixed in #271 by @ChrisJefferson - closing.
